core/sdk-core/src/main/java/software/amazon/awssdk/core/progress/listener/ProgressListener.java

```diff
@@ -0,0 +1,395 @@
+        @SdkPreviewApi
+        public interface ResponseHeaderReceived extends RequestBytesSent {
+        }
+
+        /**
+         * Additional bytes received
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link ResponseBytesReceived#request()}</li>
+         *     <li>{@link ResponseBytesReceived#progressSnapshot()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface ResponseBytesReceived extends ResponseHeaderReceived {
+        }
+
+        /**
+         * The request execution is successful.
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link ExecutionSuccess#request()}</li>
+         *     <li>{@link ExecutionSuccess#progressSnapshot()}</li>
+         *     <li>{@link ExecutionSuccess#executionSuccess()} ()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface ExecutionSuccess extends ResponseBytesReceived {
+            /**
+             * The successful completion of a request submitted to the Sdk
+             */
+            ExecutionSuccessObjectRequest executionSuccess();
+        }
+
+        /**
+         * For Expect: 100-continue embedded requests, the service returning anything other than 100 continue
+         * indicates a service error. The progress state captured indicates that no bytes are received.
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link AttemptFailureResponseBytesReceived#request()}</li>
+         *     <li>{@link AttemptFailureResponseBytesReceived#progressSnapshot()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface AttemptFailureResponseBytesReceived extends ResponseHeaderReceived {
+        }
+
+        /**
+         * The request execution attempt failed.
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link AttemptFailure#request()}</li>
+         *     <li>{@link AttemptFailure#progressSnapshot()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface AttemptFailure extends RequestPrepared {
+        }
+
+        /**
+         * The request execution failed.
+         * <p>
+         * Available context attributes:
+         * <ol>
+         *     <li>{@link ExecutionFailure#request()}</li>
+         *     <li>{@link ExecutionFailure#progressSnapshot()}</li>
+         *     <li>{@link ExecutionFailure#exception()}</li>
+         * </ol>
+         */
+        @Immutable
+        @ThreadSafe
+        @SdkPublicApi
+        @SdkPreviewApi
+        public interface ExecutionFailure extends RequestPrepared {
```

Comment:
Nit: Not used. Seems like an accident (see earlier comments)

Reply:
Yup I added this. Thanks for catching this